plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -165,98 +201,187 @@ def main():
-            if not module.check_mode:
-                group.add_member_to_group(gitlab_user_id, gitlab_group_id, access_level)
-            module.exit_json(changed=True, result="Successfully added user '%s' to the group." % gitlab_user)
-        # state as absent
-        else:
-            module.exit_json(changed=False, result="User, '%s', is not a member in the group. No change to report" % gitlab_user)
-    # in case that a user is a member
+    members = []
+    gitlab_users_access = []
+    if module.params['gitlab_user'] is not None:
+        gitlab_users = module.params['gitlab_user']
+        for glu in gitlab_users:
+            gitlab_users_access.append({'name': glu, 'access_level': access_level})
+    elif module.params['gitlab_users_access'] is not None:
+        gitlab_users_access = module.params['gitlab_users_access']
+        for i in range(len(gitlab_users_access)):
+            if gitlab_users_access[i]['access_level']:
+                gitlab_users_access[i]['access_level'] = access_level_int[gitlab_users_access[i]['access_level']]
+
+    if len(gitlab_users_access) == 1:
+        # only single user given
+        members = [group.get_member_in_a_group(gitlab_group_id, group.get_user_id(gitlab_users_access[0]['name']))]
+        if members[0] is None:
+            members = []
+    elif len(gitlab_users_access) > 1 or purge_users:
+        # list of users given
+        members = group.get_members_in_a_group(gitlab_group_id)
     else:
-        if state == 'present':
-            # compare the access level
-            user_access_level = group.get_user_access_level(members, gitlab_user_id)
-            if user_access_level == access_level:
-                module.exit_json(changed=False, result="User, '%s', is already a member in the group. No change to report" % gitlab_user)
+        # something went wrong
+        module.fail_json(msg="Please give at least one user or set purge_users true.")
+
+    changed = False
+    error = False
+    changed_users = []
+    changed_data = []
+
+    for gitlab_user in gitlab_users_access:
+        gitlab_user_id = group.get_user_id(gitlab_user['name'])
+
+        # user doesn't exist
+        if not gitlab_user_id:
+            if state == 'absent':
+                changed_users.append("user '%s' not found, and thus also not part of the group" % gitlab_user['name'])
+                changed_data.append({'gitlab_user': gitlab_user['name'], 'result': 'OK',
+                                     'msg': "user '%s' not found, and thus also not part of the group" % gitlab_user['name']})
             else:
-                # update the access level for the user
+                error = True
+                changed_users.append("user '%s' not found." % gitlab_user['name'])
+                changed_data.append({'gitlab_user': gitlab_user['name'], 'result': 'FAILED',
+                                     'msg': "user '%s' not found." % gitlab_user['name']})
```

Comment:
```suggestion
                                     'msg': "user '%s' not found." % gitlab_user['name']})
        continue
```
I think you want either a `continue` or `break` at this position. After all, the user does not exist.